In [9]:
import pandas as pd 
from pathlib import Path

PATH_NAME = Path("../data/processed/model_metrics.csv")
model_metrics_df = pd.read_csv(PATH_NAME, index_col=0)

model_metrics_df.head()

,Accuracy,Precision,Recall,ROC AUC
Logistic Regression,0.784,0.784,0.788,0.876
Random Forest,0.800,0.817,0.777,0.897
SVM,0.786,0.793,0.776,0.887


## Machine Learning Conclusions

- All three models achieved strong predictive performance, with accuracy values between 78% and 80%. Since the dataset is balanced, accuracy represents a reliable evaluation metric.
- The models also achieved high prediction and recall scores, indicating that transported passengers can be identified reliably while still capturing most actual positive cases.
- ROC AUC scores between 0.876 and 0.897 suggest that the models separate transported and non-transported passengers effectively across multiple classification thresholds.
- Among the evaluated models, Random Forest achieved the best overall performance, obtaining the highest accuracy (80%), precision (81.7%), and ROC AUC score (0.897). This suggests that nonlinear relationships and feature interactions play an important role in the dataset.
- Logistic Regression performed surprisingly well despite its simplicity, indicating that some relationships within the dataset are also approximately linear.
- Overall, the results demonstrate that passenger characteristics such as cryosleep status, spending behavior and cabin-related features contain strong predictive information regarding transportation outcome.